clinics/merge/plan.go

```diff
@@ -0,0 +1,34 @@
+package merge
+
+import (
+	"context"
+)
+
+type Plan interface {
+	PreventsMerge() bool
+}
+
+type Planner[T Plan] interface {
+	Plan(ctx context.Context) (T, error)
+}
+
+func RunPlanners[T Plan](ctx context.Context, planners []Planner[T]) ([]T, error) {
+	result := make([]T, 0, len(planners))
+	for _, planner := range planners {
+		res, err := planner.Plan(ctx)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, res)
+	}
+	return result, nil
+}
+
+func PlansPreventMerge[T Plan](plans []T) bool {
```

Comment:
Please avoid unneeded type parameters / generic functions as it makes the code harder to use. Use the interface `Plan` type directly.

```
func PlansPreventMerge(plans []Plan) bool {
```

Reply:
Unfortunately that's not possible without iterating over the arrays with concrete implementations, and creating a new array with where each structs is cast to `Plan`. For example, `[]ClinicianPlan` cannot be be passed to a function which accepts a `[]Plan` parameter.

Reply:
I see what you're saying.